### DISCUSSION TITLE
New worker does not assign tasks：Worker registered successfully, waiting for tasks

### DISCUSSION THREAD

**Discussion by ChaiWenyu1983** [filecoin-project/lotus#9696] [Storage Provider]
The new worker does not undertake the task, but the miner does.
miner：lotus-miner sealing jobs
ID        Sector  Worker    Hostname                       Task  State     Time
e015dd90  1396    5f36a8a1  chaiwenyu-System-Product-Name  C2    running   8m33.1s
00000000  1394    5f36a8a1  chaiwenyu-System-Product-Name  C2    prepared  11.6s

worker log：
2022-11-22T08:48:40.266+0800	INFO	main	lotus-worker/main.go:675	Worker registered successfully, waiting for tasks

Start command：lotus-worker run  --precommit1=false --precommit2=false --commit=true

**Comment by rjan90** [comment]
Hey @ChaiWenyu1983! If you do not want the sealing tasks to be allocated to the lotus-miner you need to disable sealing (or C2) in the configs.

You can either disable the sealing-subsystem entirely by setting `EnableSealing = false`:
```
[Subsystems]
  # type: bool
  # env var: LOTUS_SUBSYSTEMS_ENABLEMINING
  #EnableMining = true

  # type: bool
  # env var: LOTUS_SUBSYSTEMS_ENABLESEALING
  EnableSealing = false

  # type: bool
  # env var: LOTUS_SUBSYSTEMS_ENABLESECTORSTORAGE
  #EnableSectorStorage = true

  # type: bool
  # env var: LOTUS_SUBSYSTEMS_ENABLEMARKETS
  #EnableMarkets = true

  # type: string
  # env var: LOTUS_SUBSYSTEMS_SEALERAPIINFO
  #SealerApiInfo = ""

  # type: string
  # env var: LOTUS_SUBSYSTEMS_SECTORINDEXAPIINFO
  #SectorIndexApiInfo = "" 
```

Or by setting `AllowCommit=false` if you only want to disable C2 on the `lotus-miner`.
```
[Storage]
  # type: bool
  # env var: LOTUS_STORAGE_ALLOWCOMMIT
  AllowCommit = false
```

**Reply by ChaiWenyu1983** [reply]
Can't miner and worker simultaneously C2? How?
